core/startup/ErrorControlChain.php

```diff
@@ -46,14 +46,36 @@ public function setErrored($error) {
 	 */
 	public function setSuppression($suppression) {
 		$this->suppression = (bool)$suppression;
-		// Don't modify errors unless handling fatal errors, and if errors were
-		// originally allowed to be displayed.
-		if ($this->handleFatalErrors && $this->originalDisplayErrors) {
-			ini_set('display_errors', !$suppression);
+		// If handling fatal errors, conditionally disable, or restore error display
+		// Note: original value of display_errors could also evaluate to "off"
+		if ($this->handleFatalErrors) {
+			if($suppression) {
+				$this->setDisplayErrors(0);
+			} else {
+				$this->setDisplayErrors($this->originalDisplayErrors);
```

Comment:
Could there be a case where this now sets `null` for display_errors since `originalDisplayErrors` wasn't defined?